pkg/clusteragent/admission/controllers/webhook/controller_v1_test.go

```diff
@@ -1110,22 +1198,37 @@ func (f *fixtureV1) run(stopCh chan struct{}) *ControllerV1 {
 	return c
 }
 
-func (f *fixtureV1) populateWebhooksCache(webhooks ...*admiv1.MutatingWebhookConfiguration) {
+func (f *fixtureV1) populateValidatingWebhooksCache(webhooks ...*admiv1.ValidatingWebhookConfiguration) {
+	for _, w := range webhooks {
+		_, _ = f.client.AdmissionregistrationV1().ValidatingWebhookConfigurations().Create(context.TODO(), w, metav1.CreateOptions{})
+	}
+}
+
+func (f *fixtureV1) populateMutatingWebhooksCache(webhooks ...*admiv1.MutatingWebhookConfiguration) {
 	for _, w := range webhooks {
 		_, _ = f.client.AdmissionregistrationV1().MutatingWebhookConfigurations().Create(context.TODO(), w, metav1.CreateOptions{})
 	}
 }
 
-func validateV1(w *admiv1.MutatingWebhookConfiguration, s *corev1.Secret) error {
-	if len(w.Webhooks) != 3 {
-		return fmt.Errorf("Webhooks should contain 3 entries, got %d", len(w.Webhooks))
+func validateV1(validatingWebhooks *admiv1.ValidatingWebhookConfiguration, mutatingWebhooks *admiv1.MutatingWebhookConfiguration, s *corev1.Secret) error {
+	// Validate the number of webhooks.
+	if len(validatingWebhooks.Webhooks) != 0 {
+		return fmt.Errorf("validatingWebhooks should contain 1 entries, got %d", len(validatingWebhooks.Webhooks))
+	}
+	if len(mutatingWebhooks.Webhooks) != 3 {
+		return fmt.Errorf("mutatingWebhooks should contain 3 entries, got %d", len(validatingWebhooks.Webhooks))
 	}
 
-	for i := 0; i < len(w.Webhooks); i++ {
-		if !reflect.DeepEqual(w.Webhooks[i].ClientConfig.CABundle, certificate.GetCABundle(s.Data)) {
-			return fmt.Errorf("The Webhook CABundle doesn't match the Secret: CABundle: %v, Secret: %v", w.Webhooks[i].ClientConfig.CABundle, s)
+	// Validate the CA bundle for webhooks.
+	for i := 0; i < len(validatingWebhooks.Webhooks); i++ {
+		if !reflect.DeepEqual(validatingWebhooks.Webhooks[i].ClientConfig.CABundle, certificate.GetCABundle(s.Data)) {
+			return fmt.Errorf("the webhook CA bundle doesn't match the secret. CA bundle: %v, Secret: %v", validatingWebhooks.Webhooks[i].ClientConfig.CABundle, s)
+		}
```

Comment:
Here, the advantage of `testify` is that, in case of difference, the error message will show the diff between the two compared objects instead of their full raw value.

```suggestion
		require.Equal(t, validatingWebhooks.Webhooks[i].ClientConfig.CABundle, certificate.GetCABundle(s.Data))
```